odl/tomo/geometry/detector.py

```diff
@@ -0,0 +1,459 @@
+        if param not in self.params:
+            raise ValueError('parameter value {} not in the valid range {}.'
+                             ''.format(param, self.params))
+        if self.ndim == 1:
+            return float(np.linalg.norm(self.surface_deriv(param)))
+        elif self.ndim == 2:
+            return float(np.linalg.norm(np.cross(*self.surface_deriv(param))))
+        else:
+            raise NotImplementedError
+
+
+class FlatDetector(with_metaclass(ABCMeta, Detector)):
+
+    """Abstract class for flat detectors in 2 and 3 dimensions."""
+
+    def surface_measure(self, param):
+        """The constant density function of the surface measure.
+
+        Parameters
+        ----------
+        param : element of `params`
+            The parameter value where to evaluate the function
+
+        Returns
+        -------
+        meas : `float`
+            The constant density 1.0
+        """
+        if param not in self.params:
+            raise ValueError('parameter value {} not in the valid range '
+                             '{}.'.format(param, self.params))
+        return 1.0
+
+    def __repr__(self):
+        """d.__repr__() <==> repr(d)."""
+        inner_fstr = '{!r}'
+        if self.has_sampling:
+            inner_fstr += ',\n grid={grid!r}'
+        inner_str = inner_fstr.format(self.params, grid=self.param_grid)
+        return '{}({})'.format(self.__class__.__name__, inner_str)
+
+    def __str__(self):
+        """d.__str__() <==> str(d)."""
+        # TODO: prettify
+        inner_fstr = '{}'
+        if self.has_sampling:
+            inner_fstr += ',\n grid={grid}'
+        inner_str = inner_fstr.format(self.params, grid=self.param_grid)
+        return '{}({})'.format(self.__class__.__name__, inner_str)
+
+
+# TODO: rename to Flat1dDetector to be consitent with Flat2dDetector? or
+# Flat2dDetectorto to AreaDetector, or FlatLineDetector and FlatAreaDetctor
+class LineDetector(FlatDetector):
+
+    """A 1d line detector aligned with the y-axis."""
+
+    def __init__(self, params, grid=None):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        params : `Interval` or 1-dim. `IntervalProd`
+            The range of the parameters defining the detector area.
+        grid : 1-dim. `TensorGrid`, optional
+            A sampling grid for the parameter interval, in which it must
+            be contained
+        """
+        super().__init__(params, grid)
+
+        if params.ndim != 1:
+            raise ValueError('parameters {} are not 1-dimensional.'
+                             ''.format(params))
+
+    @property
+    def ndim(self):
+        """The number of dimensions of the detector."""
+        return 1
+
+    @property
+    def npixels(self):
+        """The number of pixels (sampling points)."""
+        if not self.has_sampling:
+            raise ValueError('no sampling defined.')
+        return self.param_grid.shape[0]
+
+    def surface(self, param):
```

Comment:
For all "flat" detectors this can be astractly implemented as `surface_deriv * param`.